### DISCUSSION TITLE
Rendering a single new article in an old hugo blog site

### DISCUSSION THREAD

**Discussion by harrelfe** [quarto-dev/quarto-cli#589] [Q&A] [answered]
I have an existing `hugo/netlify` blog site https://fharrell.com.  WIthout changing the overall structure or indexes I'd like to start using `quarto` for rendering.  I used the `yaml` formatting in https://quarto.org/docs/output-formats/hugo.html and ran, for the article in `my.qmd` in the `content/post/` directory, the command `quarto render my.qmd`.  This resulted in a valid looking `my.md` (although the default math rendering during a preview resulted in insertion of blank lines in place in inline equations) but the graphics files were placed in a `my_files` directory under `content/post/` instead of where `hugo` wants them to be, in `../../static/post/my_files`.  It's easy to move `my_files` but the pointers in `my.md` would be all wrong.

Is there a way to make `quarto` work in this setting where I want to create new articles as part of my existing `hugo` structure?  Note that in the existing structure `hugo` puts each article into its own directory and creates an `index` file under that directory, but what the user deals with is a markdown file (not a directory) under `content/post`.

I also noticed that a block such as `::: {.column-margin}   ... :::` does not render in the margin with the `quarto hugo` format.  One other very minor issue is that `hugo yaml tags` are not allowed to be numeric.  For example I have a `tag` of 2022 that I had to quote with `quarto`, whereas ordinary `hugo` does not require the quotes.

**Comment by jjallaire** [comment, marked as answer]
Yes, you want to use Hugo page bundles (as described here: https://quarto.org/docs/output-formats/hugo.html#creating-a-page). This basically means putting each post in its own directory.

Note that Hugo doesn't have native support for math (it requires one or more special hacks). For example, you may need to write your math within code blocks (as is done in blogdown). 

All of those margin features are particular to Quarto HTML so indeed don't work with Hugo.

**Reply by harrelfe** [reply]
I wonder if the best approach, at least for now, is to use html format and `quarto render my.qmd` with `standalone` specification in the yaml header.   The only downside is that I don't get a few hugo insertions such as a link to `disqus` at the bottom of the blog article (`hypothes.is` will probably be better for this anyway).  But I get all the `quarto` enhancements.  If you think of a downside to this approach please let me know.  Here's an example: https://fharrell.com/post/powilcoxon

I did find that I had to create a little `my.markdown` file in the same directory as `my.qmd` and `my.html` to provide the summary and tags that `hugo` needs for the article.

**Reply by jjallaire** [reply]
That definitely sounds like a reasonable way to split the difference!

**Reply by harrelfe** [reply]
I'm thrilled with the formatting. And the "extra markdown file" trick, at least so far, seems to keep `hugo` in shape with regard to article navigation, tags, etc.

**Reply by jjallaire** [reply]
Okay great, so glad we found a way to make this work.

**Comment by kjohnsen** [comment]
@harrelfe How exactly do you get Hugo to pick up the Hugo-specific frontmatter? I'm trying to use Quarto to write Hugo posts myself and find I can't use Hugo-specific frontmatter in the Quarto documents (well, unless I edit the resulting `index.html` after Quarto render, which isn't fun)

**Reply by mcanouil** [reply]
@kjohnsen See <https://quarto.org/docs/output-formats/hugo.html>.
